dataland-frontend/src/components/forms/CreateLksgDataset.vue

```diff
@@ -0,0 +1,903 @@
+        const lkSGDataControllerApi = await new ApiClientProvider(
+          assertDefined(this.getKeycloakPromise)()
+        ).getLksgDataControllerApi();
+        await lkSGDataControllerApi.postCompanyAssociatedLksgData(this.lkSGDataModel);
+        this.$formkit.reset("createLkSGForm");
+        this.isYourCompanyManufacturingCompany = "No";
+        this.listOfProductionSites = [
+          {
+            id: 0,
+            listOfGoodsOrServices: [],
+            listOfGoodsOrServicesString: "",
+          },
+        ];
+        this.idCounter = 0;
+        this.dataDate = "";
+        this.message = "Upload successfully executed.";
+        this.uploadSucceded = true;
+      } catch (error) {
+        console.error(error);
+        if (error instanceof AxiosError) {
+          this.message = "An error occurred: " + error.message;
+        } else {
+          this.message =
+            "An unexpected error occurred. Please try again or contact the support team if the issue persists.";
+        }
+        this.uploadSucceded = false;
+      } finally {
+        this.postLkSGDataProcessed = true;
+      }
+    },
+
+    /**
+     * Checks which inputs are not filled correctly
+     *
+     * @param node - single form field
+     */
+    checkCustomInputs(node: FormKitNode) {
+      const invalidElements: HTMLElement[] = [];
+      node.walk((child: FormKitNode) => {
+        // Check if this child has errors
+        if ((child.ledger.value("blocking") || child.ledger.value("errors")) && child.type !== "group") {
+          // We found an input with validation errors
+          if (typeof child.props.id === "string") {
+            const invalidElement = document.getElementById(child.props.id);
+            if (invalidElement) {
+              invalidElements.push(invalidElement);
+            }
+          }
+        }
+      }, true);
+      invalidElements.find((el) => el !== null)?.scrollIntoView({ behavior: "smooth", block: "center" });
+    },
+
+    /**
+     * Adds a new Object to the ProductionSite array
+     */
+    addNewProductionSite() {
+      this.idCounter++;
+      this.listOfProductionSites.push({
+        id: this.idCounter,
+        listOfGoodsOrServices: [],
+        listOfGoodsOrServicesString: "",
+      });
+    },
+
+    /**
+     * Remove Object from ProductionSite array
+     *
+     * @param id - the id of the object in the array
+     */
+    removeItemFromlistOfProductionSites(id: number) {
+      this.listOfProductionSites = this.listOfProductionSites.filter((el) => el.id !== id);
+    },
+
+    /**
+     * Adds a new item to the list of Production Sites Goods Or Services
+     *
+     * @param index - index of the element in the listOfProductionSites array
+     */
+    addNewItemsTolistOfProductionSites(index: number) {
+      const items = this.listOfProductionSites[index].listOfGoodsOrServicesString.split(";").map((item) => item.trim());
+      this.listOfProductionSites[index].listOfGoodsOrServices = [
```

Comment:
This line gives me an error message in the IDE. TS2322: Type 'string[]' is not assignable to type 'never[]'.   Type 'string' is not assignable to type 'never'. Did you observe it, too?

Reply:
Yes I observe it too. Will look into that now.

Reply:
Fixed that

Reply:
Florian has fixed this

Reply:
Ok, resolved